### PR TITLE
Migrate to nginx_route

### PR DESCRIPTION
**Pull request by gtrkiller**
Applicable spec: <link>

### Overview

Migrate from the ingress relation to nginx-route.

### Rationale

Keeping the charm updated with Ingress.

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

Nginx-route library replaces Ingress

### Checklist

- [X] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [X] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [X] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [X] The documentation is generated using `src-docs`
- [X] The documentation for charmhub is updated
- [X] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Comment by yanksyoon** [comment]
Woops sorry, mispressed some buttons!

File: tests/integration/any_charm.py [hunk 1]
```diff
@@ -27,21 +27,19 @@ def __init__(self, *args, **kwargs):
...
-        self.ingress = IngressRequires(
-            self,
-            {"service-hostname": self.app.name, "service-name": self.app.name, "service-port": 80},
-        )
+        require_nginx_route(charm=self, service_hostname=self.app.name, service_name=self.app.name, service_port=80)
 
-    def update_ingress(self, ingress_config):
-        """Update Ingress config.
+    def delete_nginx_route_relation_data(self, field: str) -> None:
+        """Delete one data filed from the nginx-route relation data.
```
Comment (merkata) [inline tests/integration/any_charm.py:33]: I don't understand the comment as is, can you elaborate?

Comment (merkata) [reply tests/integration/any_charm.py:33]: Or is it a typo - s/filed/field/ ?

File: tests/integration/any_charm.py [hunk 2]
```diff
@@ -27,21 +27,19 @@ def __init__(self, *args, **kwargs):
...
-            ingress_config: New Ingress configuration to be applied.
+            field: the name of the field to be deleted.
         """
-        self.ingress.update_config(ingress_config)
+        relation = self.model.get_relation("nginx-route")
+        del relation.data[self.app][field]
 
     @staticmethod
-    def start_server(port: int = 80):
+    def start_server(port: int = 8080):
```
Comment (merkata) [inline tests/integration/any_charm.py:42]: Why change this?

File: tests/integration/any_charm.py [hunk 3]
```diff
@@ -1,23 +1,23 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+# This Python script is designed to be loaded into any-charm. Some lint checks do not apply
+# pylint: disable=import-error,consider-using-with,duplicate-code
+
 """This code snippet is used to be loaded into any-charm which is used for integration tests."""
+import json
```
Comment (merkata) [inline tests/integration/any_charm.py:8]: This is currently not used and lint check fails.

File: tests/integration/any_charm.py [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
 import os
 import pathlib
 import signal
 import subprocess
-import tempfile
+from typing import Dict
```
Comment (merkata) [inline tests/integration/any_charm.py:13]: This is currently not used and lint check fails.

**Review by merkata — COMMENTED** [review]
left some comments, notably linting fails currently due to not used imports and integ fails most likely due to the change in port

File: tests/integration/any_charm.py (hunk 1 above)
Comment (gtrkiller) [reply tests/integration/any_charm.py]: It is a typo, fixed :+1:

File: tests/integration/any_charm.py (hunk 2 above)
Comment (gtrkiller) [reply tests/integration/any_charm.py]: That shouldn't be changed, I copy-pasted another any-charm file and forgot to change the port, thanks for spotting that!

**Review by amandahla — APPROVED** [review]
LGTM

**Review by nrobinaubertin — APPROVED** [review]

**Pull request merged**